### ISSUE TITLE
Returning struct from a external function gives error

### ISSUE THREAD

**Issue by saikat041** [vyperlang/vyper#2142]
### Version Information
* Vyper version: 0.2.4+commit.3eb9e7a
* OS: Ubuntu
* Python: 3.6

### What's your issue about?
If we try to return a struct from an external function we get compile error. But returning struct from internal function gives no error.
Since we can return tuple from external functions we should also be able to return struct.
```
struct Person:
    name: String[10]
    age: uint256
    
interface jsonabi:
    def test_return() -> Person: view

@external
def test_values(a: address) -> Person:
    return jsonabi(a).test_return()
```

Compiling the above code gives the following error:
```
Error compiling: try1.vy
vyper.exceptions.InvalidType: Can not get size of type, Unexpected type: 'name'
```

**Comment by fubuloubu** [comment]
btw, if you do `fixes: #2142` in your PR, it'll autoclose the issue

**Comment by saikat041** [comment]
@fubuloubu How do I edit the PR? Colud you manually close the issue?
